bridges/modules/xcm-bridge-hub/src/lib.rs

```diff
@@ -56,63 +202,1513 @@ pub mod pallet {
+
+			// check that `locations` are convertible to the `latest` XCM.
+			let bridge_origin_relative_location_as_latest: &Location =
+				bridge.bridge_origin_relative_location.try_as().map_err(|_| {
+					"`bridge.bridge_origin_relative_location` cannot be converted to the `latest` XCM, needs migration!"
+				})?;
+			let bridge_origin_universal_location_as_latest: &InteriorLocation = bridge.bridge_origin_universal_location
+				.try_as()
+				.map_err(|_| "`bridge.bridge_origin_universal_location` cannot be converted to the `latest` XCM, needs migration!")?;
+			let bridge_destination_universal_location_as_latest: &InteriorLocation = bridge.bridge_destination_universal_location
+				.try_as()
+				.map_err(|_| "`bridge.bridge_destination_universal_location` cannot be converted to the `latest` XCM, needs migration!")?;
+
+			// check `BridgeId` does not change
+			ensure!(
+				bridge_id == BridgeId::new(bridge_origin_universal_location_as_latest, bridge_destination_universal_location_as_latest),
+				"`bridge_id` is different than calculated from `bridge_origin_universal_location_as_latest` and `bridge_destination_universal_location_as_latest`, needs migration!"
+			);
+
+			// check bridge account owner
+			ensure!(
+				T::BridgeOriginAccountIdConverter::convert_location(bridge_origin_relative_location_as_latest) == Some(bridge.bridge_owner_account),
+				"`bridge.bridge_owner_account` is different than calculated from `bridge.bridge_origin_relative_location`, needs migration!"
+			);
+
+			Ok(bridge.lane_id)
+		}
+
+		/// Ensure the correctness of the state of the connected `pallet_bridge_messages` instance.
+		pub fn do_try_state_for_messages() -> Result<(), sp_runtime::TryRuntimeError> {
+			// check that all `InboundLanes` laneIds have mapping to some bridge.
+			for lane_id in pallet_bridge_messages::InboundLanes::<T, T::BridgeMessagesPalletInstance>::iter_keys() {
+				log::info!(target: LOG_TARGET, "Checking `do_try_state_for_messages` for `InboundLanes`'s lane_id: {lane_id:?}...");
+				ensure!(
+					Self::lane_to_bridge(lane_id).is_some(),
+					"Found `LaneToBridge` inconsistency for `InboundLanes`'s lane_id - missing mapping!"
+				);
+			}
+
+			// check that all `OutboundLanes` laneIds have mapping to some bridge.
+			for lane_id in pallet_bridge_messages::OutboundLanes::<T, T::BridgeMessagesPalletInstance>::iter_keys() {
+				log::info!(target: LOG_TARGET, "Checking `do_try_state_for_messages` for `OutboundLanes`'s lane_id: {lane_id:?}...");
+				ensure!(
+					Self::lane_to_bridge(lane_id).is_some(),
+					"Found `LaneToBridge` inconsistency for `OutboundLanes`'s lane_id - missing mapping!"
+				);
 			}
+
+			Ok(())
 		}
 	}
+
+	/// All registered bridges.
+	#[pallet::storage]
+	#[pallet::getter(fn bridge)]
+	pub type Bridges<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Identity, BridgeId, BridgeOf<T, I>>;
+	/// All registered `lane_id` and `bridge_id` mappings.
+	#[pallet::storage]
+	#[pallet::getter(fn lane_to_bridge)]
```

Comment:
We shouldn't add new `getter`s since [they are deprecated](https://github.com/paritytech/polkadot-sdk/issues/3326)

Reply:
fixed here: https://github.com/paritytech/polkadot-sdk/pull/4949/commits/ea9adbb3d3d717fbb38c66a8bfb675a64bf1e399